text/0026-finality-signature-gossiping.org

```diff
@@ -0,0 +1,182 @@
+#+STARTUP: inlineimages
+
+* Finality Signatures For Secure Joining Node Synchronization
+:PROPERTIES:
+:CUSTOM_ID: finality-signatures-for-secure-joining-node-synchronization
+:END:
+
+** Summary
+:PROPERTIES:
+:CUSTOM_ID: summary
+:END:
+
+CEP PR: [[https://github.com/casperlabs/ceps/pull/0026][casperlabs/ceps#0026]]
+
+Nodes joining the network will use /finality signatures/ to securely
+synchronize with the validators.  As a node synchronizes, it will get
+signatures of block hashes via the gossip network until it has greater
+than threshold =t%= (default: 33%) of the validator weight. Once a node
+has enough signatures, it can trust that block hash.  It can then
+proceed to download the next block hash.
+
+** Motivation
+:PROPERTIES:
+:CUSTOM_ID: motivation
+:END:
+
+Nodes joining the network need to get blocks from the block-chain in a
+secure manner.  If a block hash has /finality signatures/ from enough
+validators then a node can trust it. Since the security model for
+Highway is tolerant of threshold =t%= (default: 33%) of validators being
+byzantine.  If a node can see that greater than =t%= of the validator
+weight has signed a block-hash or is Byzantine, then that node can
```

Comment:
> If a node can see that greater than `t%` of the validator weight […] is Byzantine

Really? I would expect that if a node sees more than the finality threshold been Byzantine then it wouldn't trust the network at all… Doesn't it mean that there was an equivocation catastrophe and the network has to go through the manual restart?

Reply:
> Really? I would expect that if a node sees more than the finality threshold been Byzantine then it wouldn't trust the network at all… 

Well, @afck discusses this matter here: https://github.com/CasperLabs/ceps/pull/26#discussion_r545829208

While perhaps unintuitive, it doesn't violate the security parameters of the protocol.

> Doesn't it mean that there was an equivocation catastrophe and the network has to go through the manual restart?

Not necessarily no.

Reply:
If more than `ftt` (by weight) validators equivocate that is an equivocation catastrophe though.

If there is an equivocation but weight of equivocators is less than `ftt` – it's not.

In this case:
> If a node can see that greater than t% of the validator weight […] is Byzantine

it is a catastrophe. As per my undertanding.

Reply:
We may be talking past one another.

> > If a node can see that greater than t% of the validator weight […] is Byzantine

The full sentence is this:

> If a node can see that greater than =t%= of the validator
> weight has signed a block-hash or is Byzantine, then that node can
> trust that block hash.

You appear to be concerned about the case where more than =t%= of nodes are Byzantine.  In that case the security assumption is violated.

Are you trying to suggest that in that case is detected, the validator node should halt?

Reply:
Yes, I do. A validating node will stop and rely on the node operators/validators to agree on the Genesis and restart the network, as described in CEP #6. I think that a similar behaviour should be implemented for a joining node.

Reply:
I agree that a joining node should fail fast here.

Reply:
Updated in https://github.com/CasperLabs/ceps/pull/26/commits/927177407952d5d24b2a245fc9030ead7eb71295